Sources/SwiftSMTP/SMTPSocket.swift

```diff
@@ -49,7 +49,9 @@ struct SMTPSocket {
             }
         }
         let authMethod = try getAuthMethod(authMethods: authMethods, serverOptions: serverOptions, hostname: hostname)
-        try login(authMethod: authMethod, email: email, password: password)
+        if authMethod != .none {
```

Comment:
I would advocate for a dedicated `init`, since in that case, the password is useless and the email also. We do not have to specify the `authMethods` neither if I get it right in such case (must be `none`).

The functions `getAuthMethod` and `login` are useless in the `init` in the `none` case.
Maybe a minor refactoring of the connection part (`socket` and server options) into a dedicated (private) function could keep it cleared and avoid duplication.

Reply:
I see your point, and it makes sense.

Reply:
Here's a difficulty: consumers of this library won't (normally) be calling SMTPSocket(...) directly, they'll be instantiating an SMTP struct. That struct is what constructs the socket -- so, yes, having two different init methods on the socket definitely makes clear what's going on, but it means that upstream, in the SMTP struct, we'll need to gather that choice. Now, since the struct needs to support both ways of working, what do we do?

One approach would be to add a boolean such as `useAuthentication` and have it default to `true`. This would allow existing users of the library to keep their code unmodified and working correctly.

Another would be to add a second initializer to the struct, to match the two-init pattern in the socket, and then in the struct's `send` method we check to see which socket initializer to use.

Have you got a preference?